src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -309,7 +309,7 @@ create_reference_table(PG_FUNCTION_ARGS)
 	 */
 	EnsureRelationKindSupported(relationId);
 
-	workerNodeList = ActivePrimaryNodeList(ShareLock);
+	workerNodeList = ActivePrimaryCurrentDataNodeList(ShareLock);
```

Comment:
I'm wondering whether isdatanode should really apply to reference tables. We can remove a node at any time once it no longer has shards. We'll soon start replicating the reference table to the coordinator (by adding coordinator to pg_dist_node), coordinator should not be marked as a data node. We also have the notion of active/inactive.

To clear up confusion, perhaps the column should be called `hasshards`?